### PR TITLE
smlfmt: init at 1.0.0

### PR DESCRIPTION
**Pull request by Munksgaard**
###### Description of changes

Add smlfmt, a source code formatter for Standard ML source files.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/smlfmt/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  buildPhase = ''
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp smlfmt $out/bin/
+  '';
+
+  meta = {
```
Comment (bcdarwin) [inline pkgs/development/tools/smlfmt/default.nix:26]: Please add `meta.homepage`

**Review by bcdarwin — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/smlfmt/default.nix (hunk 1 above)
Comment (Munksgaard) [reply pkgs/development/tools/smlfmt/default.nix:22]: Done

File: pkgs/development/tools/smlfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    owner = "shwestrick";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-7CTfMiEvuOena5SOE0qKpFPq6ARxmkE6d+loznZNbC0=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ mlton ];
+
+  buildPhase = ''
```
[the remark this answers is not in the document]
Comment (Munksgaard) [reply pkgs/development/tools/smlfmt/default.nix]: Done

**Review by bcdarwin — APPROVED** [review]

File: pkgs/development/tools/smlfmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "shwestrick";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-7CTfMiEvuOena5SOE0qKpFPq6ARxmkE6d+loznZNbC0=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ mlton ];
```
Comment (marsam) [inline pkgs/development/tools/smlfmt/default.nix:15]: I think it only needs mlton
```suggestion
  nativeBuildInputs = [ mlton ];
```

File: pkgs/development/tools/smlfmt/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp smlfmt $out/bin/
```
Comment (marsam) [inline pkgs/development/tools/smlfmt/default.nix:19]:
```suggestion
    runHook preInstall
    install -Dm755 -t $out/bin smlfmt
    runHook postInstall
```

File: pkgs/development/tools/smlfmt/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      A custom parser and code formatter for Standard ML, with helpful error messages.
+
+      Supports SML source files (.sml, .sig, .fun, etc.) as well as MLBasis
+      compilation files (.mlb) using MLton conventions, including MLBasis path
+      maps.
+    '';
+
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ munksgaard ];
+    platforms = lib.platforms.unix;
```
Comment (marsam) [inline pkgs/development/tools/smlfmt/default.nix:34]:
```suggestion
    platforms = mlton.meta.platforms;
```

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -9740,6 +9740,16 @@
     githubId = 5047140;
     name = "Victor Collod";
   };
+  munksgaard = {
```
Comment (marsam) [inline maintainers/maintainer-list.nix:9743]: Please use a different commit for this bit, with the commit message: `maintainers: add munksgaard`

**Comment by Munksgaard** [comment]
Thank you for youor comments @marsam. I have applied your suggested changes and created a separate commit for the maintainers change.

**Pull request merged**